src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -0,0 +1,236 @@
+import { DateMixin } from '@/mixins'
+import { mapState } from 'vuex'
+import axios from '@/axios-auth'
+import { CourtOrderPoa } from '@bcrs-shared-components/court-order-poa'
+import { FormIF } from '@/interfaces'
+
+@Component({
+  computed: {
+    ...mapState(['entityIncNo'])
+  },
+  components: {
+    CourtOrderPoa
+  }
+})
+export default class AddStaffNotationDialog extends Mixins(DateMixin) {
+  $refs!: Vue['$refs'] & {
+    courtOrderPoaRef: FormIF,
+    notationFormRef: FormIF
+  }
+
+  /** Prop for the item's name of the dialog. */
+  @Prop() readonly itemName: string
+
+  /** Prop for the item's filing type. */
+  @Prop() readonly filingType: string
+
+  /** Prop to display the dialog. */
+  @Prop() readonly dialog: boolean
+
+  /** Prop to provide attachment selector. */
+  @Prop() readonly attach: string
+
+  /** Prop to require court order number regardless the plan of arrangement. */
+  @Prop() readonly courtOrderNumberRequired: boolean
+
+  /** Entity number */
+  readonly entityIncNo!: string
+
+  /** The notation text. */
+  private notation: string = ''
+
+  /** Court Order Number */
+  private courtOrderNumber:string = null
+
+  /** Whether has plan of arrangement */
+  private planOfArrangement = false
+
+  /** Whether this component is currently saving. */
+  private saving = false
+
+  /** Court Order component key */
+  private courtOrderKey = 0
+
+  /** Notation form validity */
+  private notationFormValid = false
+
+  /** Notation max length */
+  private notationMaxLength = 2000
+
+  /** Flag to enable validation in this component */
+  private enableValidation = false
+
+  private get notationRules (): Array<Function> {
+    if (this.enableValidation) {
+      return [
+        (v: string) => !!v || `Enter a ${this.itemName}`,
+        (v: string) => v.length <= this.notationMaxLength || 'Maximum characters exceeded.'
+      ]
+    } else return []
+  }
+
+  private setFileNumber (courtOrderNumber: string): void {
+    this.courtOrderNumber = courtOrderNumber
+  }
+
+  private setHasPlanOfArrangement (planOfArrangement: boolean):void {
+    this.planOfArrangement = planOfArrangement
+  }
+  /** Called when prop changes (ie, dialog is shown/hidden). */
+  @Watch('dialog')
+  private async onDialogChanged (val: boolean): Promise<void> {
+    // when dialog is shown, reset notation and validation
+    if (val) {
+      this.notation = ''
+      await Vue.nextTick()
+      this.$refs.notationFormRef.resetValidation()
+
+      // This will make CourtOrderPoa to re-render as it will have a different key every time it opens the dialog.
+      // Source: https://michaelnthiessen.com/force-re-render/
+      this.courtOrderKey++
+    }
+  }
+
+  /**
+   * Emits event to close this dialog.
+   * @param needReload Whether the dashboard needs to be reloaded.
+   */
+  @Emit('close')
+  private emitClose (needReload: boolean): void {
+    this.enableValidation = false
+  }
+
+  /** WIP */
```

Comment:
Still WIP?

Reply:
yes. Waiting for #4390

Reply:
4390 is now in RFQA. Are  you able to finish this ticket now? If not, what do you think about merging what you have so far and having a "part 2" ticket to complete it later? What do you think?

Reply:
Removed the comment and fixed the request. Now I think the most important is ok. I just need to fix css in another comment.